src/highlevel.jl

```diff
@@ -543,13 +535,18 @@ Base.length(sb::SampleBuffer) = length(sb.bufs[1])
 function Base.read(s::Stream{T}, n::Int; kwargs...) where {T}
     bufs = ntuple(_->Vector{T}(undef, n), s.nchannels)
     nread, flags, timens = _read!(s, bufs; kwargs...)
-    if nread != n
-        @info("assertion debugging", nread, n)
-        @assert nread == n
-    end
+    # By definition of read, we can allow fewer samples than requested
```

Comment:
Usually the way that these kinds of IO APIs work in Julia is that they block the task until enough data is available. May or may not be sensible here, but something to think about.

Reply:
I see...
```
  If all is true (the default), this function will block repeatedly trying to read all requested bytes, until an error or
  end-of-file occurs. If all is false, at most one read call is performed, and the amount of data returned is device-dependent.
  Note that not all stream types support the all option.
```
Then we should probably add an `all` kwarg to capture both behaviors.